clients/baseclient/client_util.go

```diff
@@ -38,6 +37,16 @@ func shouldRetry(err error) bool {
 	return false
 }
 
+func isMatching(err error) bool {
+	isMatching := strings.Contains(err.Error(), "retry is needed")
+	if isMatching {
+		return true
+	}
+	isMatching, _ = regexp.MatchString(" EOF$", err.Error())
```

Comment:
error ignoration?

Reply:
This error can occur only if the regex is invalid.